VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Middleware/AnalyticsMiddleware.cs

```diff
@@ -0,0 +1,37 @@
+using VirtualFinland.UserAPI.Helpers;
+using VirtualFinland.UserAPI.Helpers.Services;
+using VirtualFinland.UserAPI.Security.Models;
+
+namespace VirtualFinland.UserAPI.Middleware;
+
+public class AnalyticsMiddleware
```

Comment:
Oiskohan tässä kenties joku selventävä kommentti, että tämä middleware hoitaa kaksi eri asiaa:
- tracen passaaminen eteenpäin
- analytiikan kerääminen

Nuo ei ainakaan omassa päässä käänny siten, että liittyisivät suoraan toisiinsa (toinen enemmän audit trail/log hommia[?]), mutta sikäli ihan ok hoitaa samassa paikassa (olikin tuosta näköjään poisteltu joku RequestTracinMiddleware).

Reply:
Joo totta ne on omia vastuita ja hyvin voisi elellä omissa kalikoissaankin.